docs/website/blog/deploy-and-connect-to-microservice-using-helm-and-exec.md

```diff
@@ -0,0 +1,815 @@
+2. [Minikube cluster](https://minikube.sigs.k8s.io/docs/start/)
+3. [Ingress enabled on the minikube cluster](https://kubernetes.io/docs/tasks/access-application-cluster/ingress-minikube/)
+
+
+## 1. Fetch the project
+```shell
+git clone https://github.com/valaparthvi/restapi-mongodb-odo.git &&  cd restapi-mongodb-odo
+```
+
+## 2. Create namespace
+Create a namespace called `restapi-mongodb`:
+
+```shell
+odo create namespace restapi-mongodb
+```
+
+<details>
+<summary>Sample output:</summary>
+
+```shell
+$ odo create namespace
+ ✓  Namespace "restapi-mongodb" is ready for use
+ ✓  New namespace created and now using namespace: restapi-mongodb 
+```
+</details>
+
+## 3. Initialize the component
+Download the devfile to initialize an `odo` component with `odo init`.
+```shell
+odo init --devfile go --name places
+```
+
+<details>
+<summary>Sample output:</summary>
+
+```shell
+$ odo init --devfile go --name places
+  __
+ /  \__     Initializing a new component
+ \__/  \
+ /  \__/    odo version: v3.9.0
+ \__/
+
+ ✓  Downloading devfile "go" [3s]
+
+Your new component 'places' is ready in the current directory.
+To start editing your component, use 'odo dev' and open this folder in your favorite IDE.
+Changes will be directly reflected on the cluster.
+```
+</details>
+
+## 4. Modify the Devfile
+
+We will be using `odo deploy` for deploying our application, and for this we need to modify the Devfile by adding required commands and components.
+
+### Add the `commands`
+Let us begin by first adding a `deploy` command under the `commands` section.
+```yaml
+# This is the main "composite" command that will run all below commands
+- id: deploy
+  composite:
+    commands:
+      - k8s-serviceaccount-for-helm
+      - k8s-role-for-helm
+      - k8s-rolebinding-for-helm
+      - deploy-db
+      - build-image
+      - k8s-deployment
+      - k8s-service
+      - k8s-url
+    group:
+      isDefault: true
+      kind: deploy
+```
+
+`deploy` command is a composition of various other commands in the order in which we want them to be executed.
+For e.g. before deploying the database with helm, we need to ensure a service account with the required permissions (made possible by role and rolebinding) has been created;
+and so we run `k8s-serviceaccount-for-helm`, `k8s-role-for-helm` and `k8s-rolebinding-for-helm` before running `deploy-db` command.
+
+Let us now add the individual commands.
+
+We will first define the `deploy-db` command that is used to deploy the helm chart.
+
+To use an external tool such as helm or kustomize, we need to ensure 2 things:
+1. use an `exec` command; learn more [here](/docs/development/devfile#how-odo-runs-exec-commandsin-deploy-mode).
+2. the container component referenced by this command uses an image that contains the required binary.
+
+```yaml
+- id: deploy-db
+  exec:
+    commandLine: helm repo add bitnami https://charts.bitnami.com/bitnami && helm repo update && helm install mongodb bitnami/mongodb
```

Comment:
Suggestion to run `helm upgrade --install` instead, so it does not fail if we need to run it again.

```suggestion
    commandLine: helm repo add bitnami https://charts.bitnami.com/bitnami && helm repo update && helm upgrade --install mongodb bitnami/mongodb
```

Reply:
:100:
I kept running into failure when mongodb was already installed.